controllers/daemonset_test.go

```diff
@@ -275,24 +280,15 @@ var _ = Describe("daemonSetGenerator", func() {
 			).To(
 				Equal([]string{notLegitName}),
 			)
-
-			afterDeleteDaemonSetList := appsv1.DaemonSetList{}
-
-			Expect(
-				client.List(context.TODO(), &afterDeleteDaemonSetList),
-			).To(
-				Succeed(),
-			)
-
-			Expect(afterDeleteDaemonSetList.Items).To(HaveLen(1))
-			Expect(afterDeleteDaemonSetList.Items[0].Name).To(Equal(legitName))
```

Comment:
This piece of code is testing the implementation of the `client.Client`, in this case, it's the `fakeClient` and we don't want to test the client behavior, instead, we'll only rely on its return.